File: src/main/java/com/myapp/auth/TokenHandler.java
package com.myapp.auth;

import com.myapp.domain.User;
import org.springframework.security.core.userdetails.UserDetails;
import org.springframework.stereotype.Component;

import java.util.Optional;

@Component
public interface TokenHandler {

    Optional<UserDetails> parseUserFromToken(String token);

    String createTokenForUser(User user);

}

File: src/main/kotlin/com/myapp/auth/TokenHandler.kt
package com.myapp.auth

import com.myapp.domain.User
import org.springframework.security.core.userdetails.UserDetails
import org.springframework.stereotype.Component

@Component
interface TokenHandler {

    fun parseUserFromToken(token: String): UserDetails
    fun createTokenForUser(user: User): String

}
